contracts/modules/capital/Pool.sol

```diff
@@ -59,95 +83,121 @@ contract Pool is IPool, MasterAware, ReentrancyGuard {
   }
 
   constructor (
-    address[] memory assetAddresses,
-    uint8[] memory assetDecimals,
-    uint104[] memory _minAmounts,
-    uint104[] memory _maxAmounts,
-    uint16[] memory _maxSlippageRatios,
     address _master,
     address _priceOracle,
-    address _swapOperator
+    address _swapOperator,
+    address DAIAddress,
+    address stETHAddress
   ) {
+    master = INXMMaster(_master);
+    priceFeedOracle = IPriceFeedOracle(_priceOracle);
+    swapOperator = _swapOperator;
 
-    // First asset is ETH
-    assets.push(Asset(ETH, 18, false));
-
-    require(assetAddresses.length == _minAmounts.length, "Pool: Length mismatch");
-    require(assetAddresses.length == _maxAmounts.length, "Pool: Length mismatch");
-    require(assetAddresses.length == _maxSlippageRatios.length, "Pool: Length mismatch");
-    require(assetAddresses.length == assetDecimals.length, "Pool: Length mismatch");
+    // [todo] After this contract is deployed it might be worth modifying upgradeCapitalPool to
+    // copy the assets on future upgrades instead of having them hardcoded in the constructor.
 
-    for (uint i = 0; i < assetAddresses.length; i++) {
+    // The order of coverAssets should never change between updates. Do not remove the following
+    // lines!
+    coverAssets.push(Asset(ETH, 18));
+    coverAssets.push(Asset(DAIAddress, 18));
 
-      Asset memory asset = Asset(assetAddresses[i], assetDecimals[i], false);
-      require(asset.assetAddress != address(0), "Pool: Asset cannot be a zero address");
-      require(_maxAmounts[i] >= _minAmounts[i], "Pool: max < min");
-      require(_maxSlippageRatios[i] <= MAX_SLIPPAGE_DENOMINATOR, "Pool: Max slippage ratio > 1");
+    // Add investment assets
+    investmentAssets.push(Asset(stETHAddress, 18));
 
-      assets.push(asset);
-      swapDetails[asset.assetAddress].minAmount = _minAmounts[i];
-      swapDetails[asset.assetAddress].maxAmount = _maxAmounts[i];
-      swapDetails[asset.assetAddress].maxSlippageRatio = _maxSlippageRatios[i];
-    }
+    // Set DAI swap details
+    swapDetails[DAIAddress] = SwapDetails(
```

Comment:
these used to be parameters.

This is definitely nice for clarity of what's being used, but it might make some tests cumbersome if we want to the test the Pool with different values.

Reply:
We have setSwapDetails which can easily be used to test different values.